Userland/Libraries/LibAudio/MP3Loader.cpp

```diff
@@ -83,53 +82,52 @@ MaybeLoaderError MP3LoaderPlugin::seek(int const position)
         }
     }
     m_current_frame = {};
-    m_current_frame_read = 0;
     m_synthesis_buffer = {};
     LOADER_TRY(m_bit_reservoir.discard(m_bit_reservoir.used_buffer_size()));
     m_bitstream->align_to_byte_boundary();
     return {};
 }
 
-LoaderSamples MP3LoaderPlugin::get_more_samples(size_t max_samples_to_read_from_input)
+ErrorOr<Vector<FixedArray<Sample>>, LoaderError> MP3LoaderPlugin::load_chunks(size_t samples_to_read_from_input)
 {
-    FixedArray<Sample> samples = LOADER_TRY(FixedArray<Sample>::create(max_samples_to_read_from_input));
-
-    size_t samples_to_read = max_samples_to_read_from_input;
+    int samples_to_read = samples_to_read_from_input;
+    Vector<FixedArray<Sample>> frames;
     while (samples_to_read > 0) {
+        FixedArray<Sample> samples = LOADER_TRY(FixedArray<Sample>::create(1152));
```

Comment:
Weird magic number,
also used in L119
and half it in L113, L121f

Reply:
MP3 Frame size as per spec